### PR TITLE
furnace: init at 0.5.6

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Motivation for this change
https://github.com/tildearrow/furnace, a very new, FOSS & actively developed alternative to the closed-source music tracker DefleMask.

(https://github.com/NixOS/nixpkgs/issues/81815, ping @fgaz)

We need to use the vendored SDL2 because the minimum supported version is 2.0.18, which we don't have packaged (https://github.com/NixOS/nixpkgs/issues/154137).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/furnace/default.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  '';
+
+  passthru.updateScript = nix-update-script {
+    attrPath = pname;
+  };
+
+  meta = with lib; {
+    description = "Multi-system chiptune tracker compatible with DefleMask modules";
+    inherit (src.meta) homepage;
+    license = with licenses; [ gpl2Plus ];
```
Comment (fgaz) [inline pkgs/applications/audio/furnace/default.nix:85]: Where did you find the "Plus" part? I couldn't find the "or any later version" wording anywhere in the repo

Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix:85]: I asked the developer about it.

Bildschirmfoto von 2022-02-07 16-58-39

Bildschirmfoto von 2022-02-07 17-00-25

**Review by fgaz — APPROVED** [review]
The code looks good and the program runs smoothly.

And wow, this project is awesome! Mixing different chips like it's nothing

**Comment by OPNA2608** [comment]
The latest version introduced some ObjC code which breaks the Darwin build. I have yet to figure out a working fix for this :frowning:.

Bildschirmfoto von 2022-02-10 12-20-29

File: pkgs/applications/audio/furnace/default.nix [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  version = "0.5.5";
+  src = fetchFromGitHub {
+    owner = "tildearrow";
+    repo = "furnace";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-8CHRrfCe/lLPH6LYpzJUqXPSAe83v0zU18ooIDLtF84=";
+  };
+  # SDL 2.0.18+ is needed, not packages yet
+  # https://github.com/NixOS/nixpkgs/issues/154137
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/furnace/default.nix:30]: I think we can wait for https://github.com/NixOS/nixpkgs/pull/136166 which should be merged in the next days.

File: pkgs/applications/audio/furnace/default.nix [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    cp -r ../demos $out/share/furnace/
+  '';
+
+  passthru.updateScript = nix-update-script {
+    attrPath = pname;
+  };
+
+  meta = with lib; {
+    description = "Multi-system chiptune tracker compatible with DefleMask modules";
+    inherit (src.meta) homepage;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/furnace/default.nix:91]:
```suggestion
    homepage = "https://github.com/tildearrow/furnace";
```
I am not copy pasting the link together a second time.

File: pkgs/applications/audio/furnace/default.nix [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    install -m644 {../res,$out/Applications/Furnace.app/Contents}/Info.plist
+    install -m644 ../res/icon.icns $out/Applications/Furnace.app/Contents/Resources/Furnace.icns
+    install -m644 {..,$out/share/licenses/furnace}/LICENSE
+    cp -r ../papers $out/share/doc/furnace/
+    cp -r ../demos $out/share/furnace/
+  '';
+
+  passthru.updateScript = nix-update-script {
+    attrPath = pname;
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/furnace/default.nix:87]:
```suggestion
```
Not required for nix-update. If you want to keep it just resolve this.

File: pkgs/applications/audio/furnace/default.nix [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin makeWrapper;
+
+  buildInputs = [
+    fmt_8
+    libsndfile
+    vendoredSDL2
+    zlib
+  ] ++ lib.optionals withJACK [
+    libjack2
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/furnace/default.nix:54]:
```suggestion
  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
```

File: pkgs/applications/audio/furnace/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix]: Targeting staging with merged SDL2 bump now.

**Comment by OPNA2608** [comment]
Bumped to new 0.5.6 release. Spent the entire last day building a version of this merged onto staging on x68_64-linux & -darwin just to test.

**Comment by SuperSandro2000** [comment]
@ofborg build furnace

**Pull request merged**